### PR TITLE
fortran77: elseIfStatement allows comments

### PR DESCRIPTION
**Pull request by BBreiden**
elseIfStatement should also allow comments mixed with statements like the thenStatement or elseStatement.

### CODE REVIEW

File: fortran77/Fortran77Parser.g4 [hunk 1]
```diff
@@ -382,7 +382,7 @@ firstIfBlock
    ;
 
 elseIfStatement
-   : (ELSEIF | (ELSE IF)) LPAREN logicalExpression RPAREN THEN EOL? wholeStatement+
+   : (ELSEIF | (ELSE IF)) LPAREN logicalExpression RPAREN THEN EOL? commentStatement* (wholeStatement commentStatement*)+
```
Comment (KvanTTT) [inline fortran77/Fortran77Parser.g4:385]: Does it make sense to put comments to the `HIDDEN` channel (`-> channel(HIDDEN)`)? 

https://github.com/antlr/grammars-v4/blob/38857b99fbb6effa129e2d49182d3814013f8e46/fortran77/Fortran77Lexer.g4#L652-L654

In this case there won't be need of using `commentStatement` on parser level.

**Comment by BBreiden** [comment]
I have added an example for the if statement with an else-if statement.

File: fortran77/Fortran77Parser.g4 (hunk 1 above)
Comment (KvanTTT) [reply fortran77/Fortran77Parser.g4:385]: Sorry, but I don't think it's a good fix of the current problem. In the current implementation `commentStatement` is placed to certain parse rules but is not placed to others. That's why potential bugs still can be encountered. I suggest using standard technics for ignoring comments:

* Add `-> channel(HIDDEN)` to [`COMMENT`](https://github.com/antlr/grammars-v4/blob/master/fortran77/Fortran77Lexer.g4#L652) lexer rule. In this case lexer just puts tokens to another channel that is invisible for parser. `HIDDEN` channel is reserved in ANTLR and it's not neccessary to introduce a new one. Read more about lexer commands in the [ANTLR doc](https://github.com/antlr/antlr4/blob/master/doc/lexer-rules.md#channel).
* Remove `commentStatement` and its using from the parser.

**Review by KvanTTT — CHANGES_REQUESTED** [review]
Please make `COMMENT` ignorable.

File: fortran77/Fortran77Parser.g4 (hunk 1 above)
Comment (BBreiden) [reply fortran77/Fortran77Parser.g4:385]: OK. I will try to update the grammar according to your suggestions. Thanks for pointing it out to me. 👍

**Comment by BBreiden** [comment]
I redirected the comments to the `HIDDEN` channel and removed the `commentStatement`. In my setup the examples are still parsed without errors.

**Pull request merged**